src/auth/auth.ts

```diff
@@ -129,6 +130,7 @@ export const signOut = (cause: SignOutCause = 'unspecified'): void => {
 };
 
 export const signIn = async (includeBillingScope = false): Promise<OidcUser> => {
+  authStore.update((state) => ({ ...state, isInitialSignIn: true }));
```

Comment:
What does `isInitialSignIn` mean? Is it supposed to be the first time someone has signed in, or is it the first step in signing in? A comment would be helpful to explain what this is/why it's needed.

Reply:
Added a comment to explain

Reply:
How do we know that they are "first signing in"? I'm still confused about if "first/initial" means the first time they have ever signed in to Terra.

Reply:
`signIn` is called when the user clicks the `Sign In` button, so we know they're actually signing in instead of inheriting a previous session. I guess a better name would be `isNewSession`

Reply:
I'll add a comment, but basically we want to only require the user to accept a new ToS (if they have already registered) if they have come from the signIn flow. We basically do not want to force the user to accept the ToS if they open Terra in a new tab or refresh the page. So this just tells us that we have come through the signIn flow.

Reply:
Renamed to `userJustSignedIn`

Reply:
> we want to only require the user to accept a new ToS (if they have already registered) if they have come from the signIn flow. We basically do not want to force the user to accept the ToS if they open Terra in a new tab or refresh the page

Does that mean they can just refresh to bypass the new ToS?

Reply:
They're navigated to `#terms-of-service`, so refreshing won't bypass it. They could go to `#workspaces`, but that's only possible during the rolling acceptance window. If there's no rolling acceptance window, the UI will force them to accept the Terms of Service before continuing

Reply:
We've confirmed in a BEE that, once the rolling acceptance window is expired, the user cannot navigate away from the terms of service screen. Even if they manually load a different page, they will be forced to the Terms of Service screen and be required to accept it before continuing.